src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -91,5 +92,23 @@ private int GetServerMinMemory()
         {
             return configService.GetServerSmoConfig(server, configService.MinServerMemoryPropertyNumber).ConfigValue;
         }
+
+        private void UpdateServerProperties(InitializeViewRequestParams viewParams, ServerInfo serverObj)
+        {
+            if (viewParams != null)
+            {
+                ConnectionInfo connInfo = this.GetConnectionInfo(viewParams.ConnectionUri);
+                ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+                this.server = new Server(serverConnection);
+                if (GetServerMaxMemory() != serverObj.MaxServerMemory)
+                {
+                    configService.UpdateConfig(serverConnection, configService.MaxServerMemoryPropertyNumber, serverObj.MaxServerMemory);
```

Comment:
Normally we should be updating the server SMO object and then saving that - is there a reason you decided to do it this way? If doing separate queries it feels like this should be done in a transaction too so we rollback all the changes if any errors occur...

Reply:
I decided to go this route because we already have the updateConfig method in the serverConfig but I agree that when modifying multiple properties at the same time then it's not the best approach.